diff --git services/surfaceflinger/Android.mk services/surfaceflinger/Android.mk
index 72a7db7..1d742f0 100644
--- services/surfaceflinger/Android.mk
+++ services/surfaceflinger/Android.mk
@@ -21,7 +21,7 @@ LOCAL_CFLAGS:= -DLOG_TAG=\"SurfaceFlinger\"
 LOCAL_CFLAGS += -DGL_GLEXT_PROTOTYPES -DEGL_EGLEXT_PROTOTYPES
 
 ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
-    LOCAL_WHOLE_STATIC_LIBRARIES += libqc-surfaceflinger
+    #LOCAL_WHOLE_STATIC_LIBRARIES += libqc-surfaceflinger
 endif
 
 ifeq ($(TARGET_BOARD_PLATFORM), omap3)
